src/ocean/util/app/DaemonApp.d

```diff
@@ -563,7 +564,16 @@ public abstract class DaemonApp : Application,
 
     private bool statsTimer ( )
     {
-        this.onStatsTimer();
+        try
+        {
+            this.onStatsTimer();
+        }
+        catch (Exception e)
+        {
+            Stderr.formatln("Exception caught in DaemonApp.statsTimer: {}",
```

Comment:
Why stderr and not a log, for example?

Reply:
Ah, I just read the commit message that this is about out-of-disk errors.

Reply:
Yeah, I'm not even sure about `Stderr`. My assumption is that most of the daemon apps will have this either redirected to `/dev/null`, or to some circular buffer mounted on `/`, so it is unlikely that this will fail as well. Of course, I could do a try/empty catch here, just to be safe.

Reply:
I would still use the logger, as someone could have configured `Stderr` as an appender, or maybe use a network appender...

Reply:
I'm not sure that's realistic thing to assume (that one configured logger just to use `Stderr` but not a file appender), and this fix was made precisely because `log.error` was throwing.